src/Temporalio/Client/EnvConfig/ClientEnvConfig.cs

```diff
@@ -0,0 +1,489 @@
+    {
+        /// <summary>
+        /// Load client configuration from environment variables and configuration files.
+        /// </summary>
+        /// <param name="configSource">The data source to load from.</param>
+        /// <param name="configFileStrict">If true, fail if configuration file is invalid.</param>
+        /// <param name="overrideEnvVars">Environment variables to use, or null to use system environment.</param>
+        /// <returns>Loaded configuration data.</returns>
+        public static ClientEnvConfig Load(
+            DataSource? configSource = null,
+            bool configFileStrict = false,
+            IReadOnlyDictionary<string, string>? overrideEnvVars = null)
+        {
+            var runtime = Runtime.TemporalRuntime.Default.Runtime;
+            var profiles = Bridge.EnvConfig.LoadClientConfig(
+                runtime,
+                configSource,
+                configFileStrict,
+                overrideEnvVars);
+            return new ClientEnvConfig(profiles);
+        }
+
+        /// <summary>
+        /// Load client connection options directly from configuration.
+        /// </summary>
+        /// <param name="profile">Name of the profile to load. If null, "default" is used.</param>
+        /// <param name="configSource">The data source to load from.</param>
+        /// <param name="disableFile">If true, do not load from file (only from environment).</param>
+        /// <param name="disableEnv">If true, disable environment variable overrides.</param>
+        /// <param name="configFileStrict">If true, fail if configuration file is invalid.</param>
+        /// <param name="overrideEnvVars">Environment variables to use, or null to use system environment.</param>
+        /// <returns>Client connection options.</returns>
+        public static TemporalClientConnectOptions LoadClientConnectOptions(
+            string? profile = null,
+            DataSource? configSource = null,
+            bool disableFile = false,
+            bool disableEnv = false,
+            bool configFileStrict = false,
+            IReadOnlyDictionary<string, string>? overrideEnvVars = null)
+        {
+            var profileName = profile ?? "default";
+            var clientProfile = ConfigProfile.Load(
+                profileName,
+                configSource,
+                disableFile,
+                disableEnv,
+                configFileStrict,
+                overrideEnvVars);
+            return clientProfile.ToClientConnectionOptions();
+        }
+
+        /// <summary>
+        /// Convert to a record structure that can be used for TOML serialization.
+        /// </summary>
+        /// <returns>Dictionary mapping profile names to their record representations.</returns>
+        public IReadOnlyDictionary<string, ProfileRecord> ToRecord()
+        {
+            return Profiles.ToDictionary(
```

Comment:
```suggestion
        public IReadOnlyDictionary<string, ProfileRecord> ToRecord() => Profiles.ToDictionary(
```

Pedantic, but remember there's a shortcut for single-line methods